src/TraceEvent/TraceEvent.cs

```diff
@@ -2810,6 +2810,14 @@ protected internal object StateObject
         /// </summary>
         protected internal abstract void EnumerateTemplates(Func<string, string, EventFilterResponse> eventsToObserve, Action<TraceEvent> callback);
 
+        /// <summary>
+        /// Returns the set of ETWMappings between CTF event names and the corresponding ETW Provider GUID, etc.
+        /// </summary>
+        protected internal virtual IEnumerable<(string, ETWMapping)> GetCtfToETWMappings()
```

Comment:
Since this is a public API, I would like to avoid using Tuples here, and rather have an API that can sustain changes without all being a breaking change (e.g. additions).  I'm thinking something like:

 - Rename `ETWMapping` to `CtfEventMapping`.
 - Make `CtfEventMapping` a class instead of a struct to avoid copy overhead.
 - Add a first parameter to the constructor of `CtfEventMapping` for the event name.  Also add a public read-only (get) property for the name.
 - Convert the public fields into public read-only (get) properties so that the mappings are immutable.
 - Rename `GetCtfToETWMappings` to `EnumerateCtfEventMappings` to match existing naming conventions in the class.
 - Change the signature of `EnumerateCtfEventMappings` to return `IEnumerable<CtfEventMapping>`.